lib/src/rules/eol_last.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Put an end of line for non-empty files.';
+
+const _details = r'''
+
+**DO** put an end of line for non-empty files.
+
+**BAD:**
+```dart
+a {
+}
+```
+
+**GOOD:**
+```dart
+b {
+}
+
+''';
+
+class EolLast extends LintRule implements NodeLintRule {
+  EolLast()
+      : super(
+            name: 'eol_last',
```

Comment:
Consider renaming this to `eol_at_end_of_file` or `single_eol_at_end_of_file`. I think it might be easier for users to understand, and if we want to also enforce that there's a single eol marker (as was requested in https://github.com/dart-lang/linter/issues/1717) then I think it's general enough to cover that case as well.

Reply:
+1.  I prefer `eol_at_end_of_file`.